lib/Basics/ReadWriteLock.cpp

```diff
@@ -130,6 +129,22 @@ void ReadWriteLock::lockRead() {
   }
 }
 
+bool ReadWriteLock::tryLockReadFor(std::chrono::microseconds timeout) {
+  auto end_time = std::chrono::steady_clock::now() + timeout;
+  if (tryLockRead()) {
+    return true;
+  }
```

Comment:
We can avoid calculating `end_time` if we reverse the order of these operations:
```suggestion
  if (tryLockRead()) {
    return true;
  }
  auto end_time = std::chrono::steady_clock::now() + timeout;  
```